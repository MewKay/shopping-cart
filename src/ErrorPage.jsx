import { Link } from "react-router-dom";
import styles from "./ErrorPage.module.css";

const ErrorPage = () => {
  return (
    <div className={styles["container"]}>
      <h1 className={styles["status-code"]}>404</h1>
      <p className={styles["message"]}>
        Oops! The page you are looking for does not exist.
      </p>
      <Link to={"/"} className={styles["home-link"]}>
        Go Back Home
      </Link>
    </div>
  );
};

export default ErrorPage;
