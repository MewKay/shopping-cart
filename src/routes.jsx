import Home from "./pages/home/Home";
import Layout from "./layout/Layout";
import Products from "./pages/store/products-list/Products";
import Store from "./pages/store/Store";
import Cart from "./pages/cart/Cart";
import ErrorPage from "./pages/error/ErrorPage";

const routes = [
  {
    path: "/",
    element: <Layout />,
    children: [
      { index: true, element: <Home /> },
      {
        path: "store",
        element: <Store />,
      },
      {
        path: "store/:category",
        element: <Products />,
      },
      {
        path: "store/cart",
        element: <Cart />,
      },
    ],
    errorElement: <ErrorPage />,
  },
];

export default routes;
