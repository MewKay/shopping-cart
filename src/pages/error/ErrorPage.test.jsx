import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {
  createMemoryRouter,
  MemoryRouter,
  RouterProvider,
} from "react-router-dom";
import ErrorPage from "./ErrorPage";
import routes from "../../routes";

vi.mock("../home/Home", () => ({
  default: () => <>This is home</>,
}));

describe("ErrorPage component", () => {
  it("renders correctly", () => {
    const { container } = render(<ErrorPage />, { wrapper: MemoryRouter });

    expect(container).toMatchSnapshot();
  });

  it("routes back to Home page", async () => {
    const router = createMemoryRouter(routes, {
      initialEntries: ["/missing-page"],
    });
    render(<RouterProvider router={router} />);
    const user = userEvent.setup();

    const homeLink = screen.getByRole("link", { name: /Go Back Home/i });
    await user.click(homeLink);

    const homeText = await screen.findByText("This is home");
    expect(homeText).toBeInTheDocument();
  });
});
