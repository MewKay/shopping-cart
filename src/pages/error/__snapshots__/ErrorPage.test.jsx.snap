// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage component > renders correctly 1`] = `
<div>
  <div
    class="_container_e373f7"
  >
    <h1
      class="_status-code_e373f7"
    >
      404
    </h1>
    <p
      class="_message_e373f7"
    >
      Oops! The page you are looking for does not exist.
    </p>
    <a
      class="_home-link_e373f7"
      href="/"
    >
      Go Back Home
    </a>
  </div>
</div>
`;
